src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -653,6 +661,155 @@ MultiClientCopyData(int32 connectionId, int32 fileDescriptor)
 
 
 /*
+ * MultiClientCreateWaitInfo creates a WaitInfo structure, capable of keeping
+ * track of what maxConnections connections are waiting for; to allow
+ * efficiently waiting for all of them at once.
+ *
+ * Connections can be added using MultiClientAddWait(). All added connections
+ * can then be waited upon together using MultiClientWait().
+ */
+WaitInfo *
+MultiClientCreateWaitInfo(int maxConnections)
+{
+	WaitInfo *waitInfo = palloc(sizeof(WaitInfo));
```

Comment:
Shouldn't you use `palloc0` here? The boolean fields aren't initialized to anything, seems better to just `palloc0` unless there's a clear reason not to.